### ISSUE TITLE
Require email verification

### ISSUE THREAD

**Issue by alvolex** [NodeBB/NodeBB#10872]
### Description

**Issue:**
A user can currently register multiple accounts with the same e-mail as long as they don't verify the email on any of the accounts.
Even with the “Require new users to specify an email address” option ticked, there is nothing that requires the user to verify the email.
Another issue that is related to this is that users with unverified emails are not able to reset their password, which could potentially lead to dead accounts.

**Proposed solution:**
Add a new optional setting that would force the user to verify the account during the registration process, possibly by sending them an email with a code that they need to enter to complete the registration.

This would prevent users from getting locked out of their own accounts if they were to forget to verify their email and/or managed to mess up their password during registration. This would also solve the issue of having multiple users with the same email (while the email is unverified).

**Comment by julianlam** [comment]
> A user can currently register multiple accounts with the same e-mail as long as they don't verify the email on any of the accounts.

This is by design, although I should point out very explicitly that **if you don't verify your email, that email is not on your account**. Only once the email is verified, is the email added to the account. This is to prevent issues where users register using emails that they do not actually own.

If you want to deny access to the forum until users have verified their emails, you need to change the privileges on each category. Remove all the privileges from `registered-users`, and add them to `verified-users`.

**Comment by twcctz200** [comment]
good luck to you
